### PR TITLE
Convert array results to classes

### PR DESCRIPTION
**Pull request by gpenverne**
- Method get($id) will return instance of class
- Getter and Setter using __call
- all() will return array of instances
- toArray() will return raw array
- retrieve the int using the getId() method

### CODE REVIEW

File: src/TrustFully/Api/AbstractApi.php [hunk 1]
```diff
@@ -10,6 +10,14 @@
  */
 abstract class AbstractApi implements ApiInterface
 {
+    const HYDRA_MEMBER_KEY = 'hydra:member';
+
+    const HYDRA_TYPE_KEY = '@type';
+
+    const HYDRA_TYPE_ID = '@id';
```
Comment (kbsali) [inline src/TrustFully/Api/AbstractApi.php]: HYDRA_ID

File: src/TrustFully/Api/AbstractApi.php [hunk 2]
```diff
@@ -10,6 +10,14 @@
...
+
+    const HYDRA_TYPE_COLLECTION = 'hydra:Collection';
```
Comment (kbsali) [inline src/TrustFully/Api/AbstractApi.php]: HYDRA_COLLECTION

File: src/TrustFully/Api/AbstractApi.php [hunk 3]
```diff
@@ -35,7 +49,17 @@ public function __construct(ClientInterface $client)
      */
     public function all(array $params = [])
     {
-        return $this->client->get(sprintf('/%s', $this->endPoint), $params);
+        $data = $this->client->get(sprintf('/%s', $this->endPoint), $params);
+        if (is_array($data) && isset($data[self::HYDRA_MEMBER_KEY]) && $data[self::HYDRA_TYPE_KEY] == self::HYDRA_TYPE_COLLECTION) {
```
Comment (kbsali) [inline src/TrustFully/Api/AbstractApi.php]: == -> ====

File: src/TrustFully/Api/AbstractApi.php [hunk 4]
```diff
@@ -45,7 +69,58 @@ public function all(array $params = [])
...
+
+    /**
+     * @param string $method
+     * @param mixed  $args
+     *
+     * @return mixed
+     */
+    public function __call($method, $args)
+    {
+        if (true == preg_match('%get(.*)%', $method, $matches)) {
```
Comment (kbsali) [inline src/TrustFully/Api/AbstractApi.php]: == -> ===

File: src/TrustFully/Api/AbstractApi.php (hunk 1 above)
Comment (gpenverne) [inline src/TrustFully/Api/AbstractApi.php]: :+1:

File: src/TrustFully/Api/AbstractApi.php (hunk 2 above)
Comment (gpenverne) [inline src/TrustFully/Api/AbstractApi.php]: :+1:

File: src/TrustFully/Api/AbstractApi.php (hunk 3 above)
Comment (gpenverne) [inline src/TrustFully/Api/AbstractApi.php]: :+1:

File: src/TrustFully/Api/AbstractApi.php (hunk 4 above)
Comment (gpenverne) [inline src/TrustFully/Api/AbstractApi.php]: :+1:

**Pull request merged**